script/archives/archive_stake_account_history.rb

```diff
@@ -0,0 +1,13 @@
+#frozen_string_literal: true
+
+require File.expand_path('../../config/environment', __dir__)
+
+ARCHIVE_TIME = {
+  mainnet: 2.months.ago,
+  testnet: 2.months.ago,
+  pythnet: 2.months.ago,
+}.freeze
+
+NETWORKS.each do |network|
+  StakeAccountHistory.archive_due_to(date_to: ARCHIVE_TIME[network.to_sym], network: network, destroy_after_archive: false)
```

Comment:
We can add extra argument to script which determines whether `destroy_after_archive` is true or not.